src/cryptonote_core/miner.cpp

```diff
@@ -408,5 +528,313 @@ namespace cryptonote
+  {
+    #if defined(WIN32) || defined(_WIN32) || defined(__WIN32)
+
+    LOG_PRINT_RED_L0("background mining not supported on windows ... yet");
+    return false;
+
+    #else
+
+    const std::string STR_CPU("cpu");
+    const std::size_t STR_CPU_LEN = STR_CPU.size();
+    const std::string STAT_FILE_PATH = "/proc/stat";
+    const std::string PROC_FILE_PATH = "/proc/self/stat";
+
+    if( !epee::file_io_utils::is_file_exist(STAT_FILE_PATH) )
+    {
+      LOG_PRINT_RED_L0(STAT_FILE_PATH << " file does not exist");
+      return false;
+    }
+
+    std::ifstream stat_file_stream(STAT_FILE_PATH);
+    if( stat_file_stream.fail() )
+    {
+      LOG_PRINT_RED_L0("failed to read " << STAT_FILE_PATH);
+      return false;
+    }
+
+    cpud.update_time = misc_utils::get_tick_count();
+    std::string line;
+    std::getline(stat_file_stream, line);
+    std::istringstream stat_file_iss(line);
+    std::string cpu_label;
+
+    stat_file_iss >> cpu_label;
+    for(size_t i = 0; i < CPU_DATA_OVERALL_TIMES_COUNT; ++i)
+    {
+      stat_file_iss >> cpud.overall_cpu_times[i];
+    }
+
+
+    std::ifstream proc_file_stream(PROC_FILE_PATH);
+    if( proc_file_stream.fail() )
+    {
+      LOG_PRINT_RED_L0("failed to read " << PROC_FILE_PATH);
+      return false;
+    }
+
+    std::getline(proc_file_stream, line);
+    std::istringstream proc_file_iss(line);
+    std::string tmp;
+    for(size_t i = 0; i < 13; ++i)
+    {
+      proc_file_iss >> tmp;
+      //LOG_PRINT_RED_L0(" ( " << i << ") --> " << tmp);
+      //proc_file_iss.ignore(65536, ' ');
+    }
+
+    float utime, stime;
+    proc_file_iss >> utime;
+    proc_file_iss >> stime;
+    //LOG_PRINT_L0("CUTIME --> " << utime << ", SUTIME --> " << stime);
+    float clk_tck = sysconf(_SC_CLK_TCK) * 1.f;
+    //LOG_PRINT_L0(" " << utime << " / " << clk_tck);
+    //LOG_PRINT_L0(" " << stime << " / " << clk_tck);
+    cpud.process_cpu_time = ((uint64_t)  ( (utime / clk_tck + stime / clk_tck) / .01 ) );
+    //LOG_PRINT_L0("process_cpu_time --> " << cpud.process_cpu_time);
+
+    /*
+    if( m_cpu_count <= 0 )
+    {
+      uint8_t cpu_count = 0;
+      while ( std::getline(stat_file_iss, line) )
+      {
+        LOG_PRINT_RED_L0("line = " << line);
+        if( !line.compare(0, STR_CPU_LEN, STR_CPU) )
+        {
+          stat_file_iss >> cpu_label;
+          LOG_PRINT_RED_L0("cpu_label = " << cpu_label);
+          if( cpu_label.size() >= STR_CPU_LEN )
+          {
+            // so this is a line starting with cpu, but not the total line
+            ++cpu_count;
+          }
+        }
+      }
+
+      m_cpu_count = cpu_count;
+      LOG_PRINT_RED_L0("m_cpu_count = " << cpu_count);
+    }
+    */
+
+    return true;
+
+    #endif
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::should_i_start_mining(const cpu_data& cpu_data_start, const cpu_data& cpu_data_end, bool& start_mining)
```

Comment:
These double bools are confusing. Can't this function just return true when mining should start, and fail in other situations? The function is already logging errors.